src/lib/components/Actions/index.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview Element that renders configurable per-page actions.
+ */
+
+import {html} from "lit-element";
+import {BaseElement} from "../BaseElement";
+
+function isWebShareSupported() {
+  if (!("share" in navigator)) {
+    return false;
+  }
+
+  // Ensure that the user would be able to share a reference URL.
+  // This is part of Web Share Level 2, so feature-detect it:
+  // https://bugs.chromium.org/p/chromium/issues/detail?id=903010
+  if ("canShare" in navigator) {
+    const url = `https://${window.location.hostname}`;
+    return navigator.canShare({url});
```

Comment:
I think I originally imagined it to be a global. It's not really related to the class. It could be in a misc helper library.

Also, we're not _setting_ canShare. It's an extension in Level 2 of the API which asks the browser _synchronously_ whether a share would be possible (as `navigator.share` doesn't tell you that—it only _enacts_ a possible share). I clearly need to make that more obvious :)

Reply:
ah I see.

yeah might make sense to pull it out of this class and put it into `lib/utils`.